cadc-rest/src/main/java/ca/nrc/cadc/rest/RestServlet.java

```diff
@@ -488,29 +519,24 @@ void setAuthenticateHeaders(Subject subject, SyncOutput out, NotAuthenticatedExc
                 log.debug("LocalAuthority -- not found: " + Standards.SECURITY_METHOD_OAUTH, notSupported);
             }
             
-            // TODO: mechanism to enable this rather than always set it
+            try {
+                // header for delegated x509 client proxy certificates
+                URI cdpProxyServiceURI = getLocalServiceURI(Standards.CRED_PROXY_10);
+                URL cdpProxyURL = regClient.getServiceURL(cdpProxyServiceURI, Standards.CRED_PROXY_10, AuthMethod.PASSWORD);
```

Comment:
Nice: working this out took care of that TODO to find a mechanism :-)

If you didn't use the shortcut `regClient.getServiceURL` this could be written look for CRED_PROXY_10 with either #BasicAA or #tls-with-password securityMethod instead of just basic... then if we changed /cred to use #tls-with-password, or someone else deployed differently, this would still work as expected. It would be something like:

```
Capabilities caps = regClient.getCapabilities(cdpProxyServiceURI);
Capability c = caps.findCapability(Standards.CRED_PROXY_10);
for (Interface i : c.getInterfaces()) {
    // find interface for suitable securityMethod to use as standard_id in challenge
}
```
The fact that this would work validates our approach because in principle we don't know what /cred provides but can extract it from caps and put it into challenge anyway.